kotlin/perform-valve-isolation-trace/src/main/java/com/esri/arcgisruntime/sample/performvalveisolationtrace/MainActivity.kt

```diff
@@ -0,0 +1,295 @@
+        elementFeaturesFuture.addDoneListener {
+          try {
+            val startingLocationFeatures = elementFeaturesFuture.get()
+            if (startingLocationFeatures.isNotEmpty()) {
+              // get the geometry of the first feature for the starting location as a point
+              (startingLocationFeatures[0].geometry as? Point)?.let { startingLocationGeometryPoint ->
+
+                // create a graphic for the starting location and add it to the graphics overlay
+                val startingLocationGraphic =
+                  Graphic(startingLocationGeometryPoint, startingPointSymbol)
+                startingLocationGraphicsOverlay.graphics.add(startingLocationGraphic)
+
+                // set the map's viewpoint to the starting location
+                mapView.setViewpointAsync(Viewpoint(startingLocationGeometryPoint, 3000.0))
+
+                // populate the spinner with utility categories as the data and their names as the text
+                spinner.adapter = object : ArrayAdapter<UtilityCategory>(
+                  this,
+                  R.layout.spinner_text_item,
+                  networkDefinition.categories
+                ) {
+                  override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
+                    val spinnerItem = LayoutInflater.from(this@MainActivity)
+                      .inflate(R.layout.spinner_text_item, parent, false)
+                    spinnerItem.textView.text = (getItem(position) as UtilityCategory).name
+                    return spinnerItem
+                  }
+
+                  override fun getDropDownView(
+                    position: Int,
+                    convertView: View?,
+                    parent: ViewGroup
+                  ): View = getView(position, convertView, parent)
+                }
+
+                trace_button.setOnClickListener {
+                  fab.isExpanded = false
+                  performTrace(utilityNetwork, traceConfiguration, startingLocation)
+                }
+              }
+            } else {
+              val message = "Starting location features not found."
+              Log.i(TAG, message)
+              Toast.makeText(
+                this,
+                message,
+                Toast.LENGTH_LONG
+              ).show()
+            }
+          } catch (e: Exception) {
+            val error = "Error loading starting location feature: ${e.message}"
+            Log.e(TAG, error)
+            Toast.makeText(this, error, Toast.LENGTH_LONG).show()
+          }
+        }
+      } else {
+        val error = "Error loading utility network: ${utilityNetwork.loadError}"
+        Log.e(TAG, error)
+        Toast.makeText(this, error, Toast.LENGTH_LONG).show()
+      }
+    }
+  }
+
+  /**
+   * Performs a valve isolation trace according to the defined trace configuration and starting location.
+   * Selects the resulting features on the map.
```

Comment:
```suggestion
  /**
   * Performs a valve isolation trace according to the defined trace configuration and starting location, and selects the resulting features on the map.
```

Reply:
this was mainly done to not overflow the line, but I'm happy to have it as one line